x-pack/legacy/plugins/watcher/public/models/watch/threshold_watch.js

```diff
@@ -206,7 +206,7 @@ export class ThresholdWatch extends BaseWatch {
     Array.from(Array(comparators[this.thresholdComparator].requiredValues)).forEach((value, i) => {
       const key = `threshold${i}`;
       errors[key] = [];
-      if (!this.threshold[i]) {
+      if (typeof this.threshold[i] === 'undefined' || this.threshold[i] === '') {
```

Comment:
Not entirely sure how `comparators[this.thresholdComparator]` works and what the known values for `requiredValues` is - but if it ever changes and assigns `null` this logic won't work anymore. I'd suggest a slightly more defensive check:

```ts
  if (
  this.threshold[i] == null // terse way of checking for `null` and `undefined` at the same time
|| this.threshold[i] === '') {
```

Reply:
good point - updated the code!